rest/route/version.go

```diff
@@ -63,6 +64,59 @@ func (vh *versionHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.NewJSONResponse(versionModel)
 }
 
+////////////////////////////////////////////////////////////////////////
+//
+// PATCH /rest/v2/versions/{version_id}
+
+type versionPatchHandler struct {
+	Activated *bool `json:"activated"`
+
+	versionId string
+}
+
+func makePatchVersion() gimlet.RouteHandler {
+	return &versionPatchHandler{}
+}
+
+func (vh *versionPatchHandler) Factory() gimlet.RouteHandler {
+	return &versionPatchHandler{}
+}
+
+// Parse fetches the versionId from the http request.
+func (vh *versionPatchHandler) Parse(ctx context.Context, r *http.Request) error {
+	if err := utility.ReadJSON(r.Body, vh); err != nil {
+		return errors.Wrap(err, "reading body")
+	}
+	if vh.Activated == nil {
+		return gimlet.ErrorResponse{
+			Message:    "Must set 'activated'",
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+
+	vh.versionId = gimlet.GetVars(r)["version_id"]
+	if vh.versionId == "" {
+		return errors.New("request data incomplete")
+	}
+	return nil
+}
+
+// Run calls the data model.VersionFindOneId function
+func (vh *versionPatchHandler) Run(ctx context.Context) gimlet.Responder {
+	u := MustHaveUser(ctx)
+	if err := dbModel.SetVersionActivation(vh.versionId, utility.FromBoolPtr(vh.Activated), u.Id); err != nil {
+		state := "inactive"
```

Comment:
I think this messaging is pretty outdated; I think the error at 112 should just actually return the error

Reply:
done